### ISSUE TITLE
yarn run --non-interactive prompts and errors

### ISSUE THREAD

**Issue by RoBorg** [yarnpkg/yarn#5655]
**Do you want to request a *feature* or report a *bug*?**: Bug

**What is the current behavior?**
`yarn run --non-interactive` gives errors and is interactive

**If the current behavior is a bug, please provide the steps to reproduce.**

```
$ yarn run --non-interactive
yarn run v1.6.0
error No command specified.
info Commands available from binary scripts: acorn, --SNIP--
info Project commands
   - dev
      gulp watch
   - prod
      gulp --production
question Which command would you like to run?:
```

**What is the expected behavior?**

1. The `error No command specified.` message should not be printed as `yarn run` with no command is supposed to list available commands
2. The `question Which command would you like to run?:` should not appear with `--non-interactive` specified

**Please mention your node.js, yarn and operating system version.**
node v8.9.4, yarn v1.6.0 (same on v1.5.1), Windows 10

**Comment by BhaveshSGupta** [comment]
@rally25rs, I would love to try to solve this.

**Comment by paul-soporan** [comment]
Closed in #5694.
